src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -2076,7 +2076,8 @@ GetSingleStackTraceClosure::do_thread(Thread *target) {
 
 void
 GetSingleStackTraceClosure::do_vthread(Handle target_h) {
-  assert(_target_jt == nullptr || _target_jt->vthread() == target_h(), "sanity check");
+  // use jvmti_vthread() as vthread() can be outdated
```

Comment:
The only reason I can see of why just using vthread() doesn't work is because of the case where we are in a temporary switch to carrier thread. So maybe change comment to be: "use jvmti_vthread() instead of vthread() as target could have temporary changed identity to carrier thread (see VirtualThread.switchToCarrierThread)". Same in the other places.